### ISSUE TITLE
Repo: setup a GH action to help validate and educate PR title

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#5110]
### Suggestion

Spinning off of #4981 (/#4962)

We used to have https://github.com/zeke/semantic-pull-requests setup and working as a CI step, however it appears to no longer work. I'm not sure why!

I've double checked and we have the app installed in the repo so it *should* just work?

<details>
<summary>Maybe because we don't have a config? It might be worth trying to configure it</summary>

File: `.github/semantic.yml`
```yaml
enabled: true
titleOnly: true
scopes:
  - ast-utils
  - ...
types:
  - feat
  - fix
  - docs
  - chore
  - test
  - revert
```

</details>

Otherwise, given that we're all-in on GH actions - we could just setup a GH action for this!
First hit on google is https://github.com/amannn/action-semantic-pull-request.
This might be the better action so we can reduce our app surface area and better monitor the things we're using.
